test/e2e/user_workload_monitoring_test.go

```diff
@@ -681,6 +703,122 @@ func assertTenancyForMetrics(t *testing.T) {
 	if err != nil {
 		t.Fatalf("the account has access to the rules endpoint of Thanos querier: %v", err)
 	}
+
+	for _, tc := range []struct {
+		role               rbacv1.Role
+		expectNotOKOnQuery bool
+		desc               string
+		method             string
+	}{
+
+		{
+			role: rbacv1.Role{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "tenancy-test-metrics",
+				},
+				Rules: []rbacv1.PolicyRule{
+					{
+						APIGroups: []string{"metrics.k8s.io"},
+						Resources: []string{"pods"},
+						Verbs:     []string{"get"},
+					},
+				},
+			},
+			method:             http.MethodPost,
+			expectNotOKOnQuery: true,
+			desc:               "should disallow POST queries to the endpoint for SA with no create permission",
+		},
+		{
+			role: rbacv1.Role{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "tenancy-test-metrics",
+				},
+				Rules: []rbacv1.PolicyRule{
+					{
+						APIGroups: []string{"metrics.k8s.io"},
+						Resources: []string{"pods"},
+						Verbs:     []string{"get"},
+					},
+				},
+			},
+			method: http.MethodGet,
+			desc:   "should allow GET queries to the endpoint for SA with get permission",
+		},
+		{
+			role: rbacv1.Role{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "tenancy-test-metrics",
+				},
+				Rules: []rbacv1.PolicyRule{
+					{
+						APIGroups: []string{"metrics.k8s.io"},
+						Resources: []string{"pods"},
+						Verbs:     []string{"get", "create"},
+					},
+				},
+			},
+			method: http.MethodPost,
+			desc:   "should allow POST queries to the endpoint for SA with get and create permission",
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			var cleanupFn framework.CleanUpFunc
+
+			// Create a role binding for the test SA.
+			err = framework.Poll(time.Second, time.Minute, func() error {
+				cleanupFn, err = f.CreateRoleBindingFromTypedRole(userWorkloadTestNs, testAccount, &tc.role)
+				return err
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Remove associated artifacts.
+			defer func() {
+				if err := cleanupFn(); err != nil {
+					t.Fatal(err)
+				}
+			}()
+
+			// Forward the tenancy port.
+			host, cleanUp, err := f.ForwardPort(t, f.Ns, "thanos-querier", 9092)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer cleanUp()
+
+			// Create a Prometheus client with the test SA token.
+			client := framework.NewPrometheusClient(
+				host,
+				token,
+				&framework.QueryParameterInjector{
+					Name:  "namespace",
+					Value: userWorkloadTestNs,
+				},
+			)
+
+			resp, err := client.Do(tc.method, "/api/v1/query?namespace="+userWorkloadTestNs+"&query=up", nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer resp.Body.Close()
+			// Body: {"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"up",...},"value":[1695582946.784,"1"]}]}}
+			respBodyBytes, err := io.ReadAll(resp.Body)
```

Comment:
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-monitoring-operator/2357/pull-ci-openshift-cluster-monitoring-operator-release-4.14-e2e-agnostic-operator/1793117774470451200#1:build-log.txt%3A99

I'll patch this up.